theseus/embodied/kinematics/kinematics_model.py

```diff
@@ -0,0 +1,112 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import abc
+from typing import Dict, Optional, Union
+
+import differentiable_robot_model as drm
+import torch
+
+from theseus import Variable
+from theseus.geometry import SE3, LieGroup, Point2, Vector
+
+RobotModelInput = Union[torch.Tensor, Vector]
+
+
+class KinematicsModel(abc.ABC):
+    def __init__(self):
+        pass
+
+    @abc.abstractmethod
+    def forward_kinematics(self, robot_pose: RobotModelInput) -> Dict[str, LieGroup]:
+        pass
+
+    @abc.abstractmethod
+    def dim(self) -> int:
+        pass
+
+
+class IdentityModel(KinematicsModel):
+    def __init__(self):
+        super().__init__()
+
+    def forward_kinematics(self, robot_pose: RobotModelInput) -> Dict[str, LieGroup]:
+        if isinstance(robot_pose, Point2) or isinstance(robot_pose, Vector):
+            assert robot_pose.dof() == 2
+            return {"state": robot_pose}
+        raise NotImplementedError(
+            f"IdentityModel not implemented for pose with type {type(robot_pose)}."
+        )
+
+    def dim(self) -> int:
+        return 1
```

Comment:
1 -> 2?

Reply:
This object was implemented by @luisenp, can you confirm that the return value should be 2 insteaf?

Reply:
I think it should be 2, but this was set to 1 quite a long time ago, it seems. I don't remember why, but maybe it was to do [this](https://github.com/facebookresearch/theseus/blob/main/theseus/embodied/collision/collision.py#L73), which seems incorrect? As in, collision dim should just be hardcoded to 1?

BTW, do we actually need this dim function?

Reply:
Ah for now hardcoding collision dim to one is okay when it is identity model. Otherwise it should be the number of body points you are checking collision for.

Reply:
Then in this case it seems that `dim()` for kinematics models is actually representing the number of body points? Should we rename this method?

Reply:
Not sure yet how we want to add body points, inside kinematics class or as separate struct/class. Either way it should be a property related to that. If that was the only use for dim then maybe we can remove it for now.

Reply:
Should I simply remove the dim method from `IdentityModel` and `UrdfRobotModel`, and also downstream methods in `Collision2D` and `EffectorObjectContactPlanar` that uses it?

Reply:
Removed and tested in #156